src/box/lua/connpool.lua

```diff
@@ -125,6 +129,51 @@ local function connect_to_candidates(candidates)
     return connected_candidates
 end
 
+local acquire_any_cond = nil
+
+-- The method starts connecting to the specified set of
+-- instances and returns the first one available matching
+-- the specified dynamic requirements.
+--
+-- Note: if the specified mode is 'alive' the connection
+-- returned may not have the `mode()` method available.
+local function acquire_any_candidate(instance_names, opts)
+    assert(type(opts) == 'table' and opts.mode)
+    assert(opts.mode == 'ro' or opts.mode == 'rw' or opts.mode =='alive')
+    local time_connect_end = clock.monotonic() + WATCHER_TIMEOUT
```

Comment:
Lets name it something like `connect_deadline` or something similar.

Reply:
Applied. Thank you.

Reply:
Don't know what occurs here, but it seems the old name still there.

Reply:
I've dropped such unrelated changes after applying @Totktonada PoC. See here https://github.com/tarantool/tarantool/pull/10438#discussion_r1862483309.

Reply:
Seems resolved, so I'm going to close this discussion.